user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js

```diff
@@ -1216,15 +1240,97 @@ define([
         obj: $target,
         time: (new Date()).getTime()
       });
+      $(".folder.selected").children(".element").attr("tabindex",-1).attr("aria-selected",false);
       $(".folder.selected").removeClass("selected");
       $(".folder.secondarySelected").removeClass("secondarySelected");
       $target.addClass("selected");
+      $target.children(".element").attr("tabindex",0).attr("aria-selected",true);
       //deselect any files
       $(".file.selected").removeClass("selected");
       depth = $target.attr("path").split("/").length;
       event.stopPropagation();
     },
 
+    keyDownFolder: function (event) {
+      let keyCode = event.which || event.keyCode;
+      if (keyCode === 13 || keyCode === 32){
+        // ENTER , SPACE
+        this.clickFolder(event);
+      }else if (keyCode === 40){
+        // DOWN Arrow
+        let nextElement = this.getNextElementToFocus($(event.currentTarget));
+        if(nextElement !== undefined){
+         nextElement.focus();
+        }
+      }else if (keyCode === 38){
+        // UP Arrow
+        let prevElement = this.getPreviousElementToFocus($(event.currentTarget));
+        if (prevElement !== undefined){
+          prevElement.focus();
+        }
+      }else if (keyCode === 39){
+        // RIGHT Arrow
+        if (!$(event.currentTarget).parent().hasClass("open")){
+          this.expandFolder(event);
+        }
+      }else if(keyCode === 37){
+        // LEFT Arrow
+        if ($(event.currentTarget).parent().hasClass("open")){
+          this.expandFolder(event);
+        }
+      }
+      event.stopPropagation();
+    },
+
+    getNextElementToFocus : function (currentElement) {
+      let lastFolder = $("#fileBrowserFolders").find(".folder").last();
+      if (currentElement.parent().hasClass("open")){
+        let firstChildFolder = currentElement.next().children().first();
+        if (firstChildFolder.length !== 0) {
+          return firstChildFolder.children(".element");
+        }
+        return this.getNextAvailableElement(currentElement.parent(),lastFolder);
+      }else{
+        return this.getNextAvailableElement(currentElement.parent(),lastFolder);
+      }
+    },
+
+    getNextAvailableElement: function (currentFolder, lastFolder) {
```

Comment:
Returning `null` when no next element exists would be more natural, maintaining the return type in the realm of objects (`typeof` equal to "object"; in JsDoc terms it would be listed as `?object`).